restapi.go

```diff
@@ -82,6 +91,12 @@ func (s *Session) RequestWithLockedBucket(method, urlStr, contentType string, b
 		return
 	}
 
+	if optHeaders != nil {
+		for opt := range optHeaders {
```

Comment:
If we're going to use a range-for loop, perhaps just going straight for the value version rather than the index would be a tad cleaner?